### PR TITLE
[STF] Fix self tests

### PR DESCRIPTION
**Pull request by m273d15**
As a preparation for the SWTBot update (#587), I fixed the currently failing STF self-test.

The failures were caused by:
* A few small changes in Eclipse
* We changed the context menu and project sharing entry
* The assumption that the follow-mode also closes editors

### CODE REVIEW

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- stf/test/saros/stf/test/stf/contextmenu/ContextMenuShareProjectsTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5680267&bid=18594258)

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- stf/test/saros/stf/test/stf/contextmenu/ContextMenuShareProjectsTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5680267&bid=18594258)

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- stf/test/saros/stf/test/stf/contextmenu/ContextMenuShareProjectsTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5680267&bid=18594258)

**Comment by m273d15** [comment]
Some tests are still flaky. With using the retry-approach (as in STF) all tests are [successful](https://github.com/saros-project/saros/runs/945218337?check_suite_focus=true). 

> These tree tests consistently fail for me:

I will try to reproduce this.

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- stf/test/saros/stf/test/stf/contextmenu/ContextMenuShareProjectsTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5680267&bid=18594258)

**Comment by m273d15** [comment]
I was not able to reproduce these three failures (using the eclipse launch configs). Probably, your system/wm handles shortcuts differently, but this would only explain `#pressShortCutSave`.

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- stf/test/saros/stf/test/stf/contextmenu/ContextMenuShareProjectsTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5680267&bid=18594258)

**Comment by srossbach** [comment]
There were always issues with the PackageExplorer View when using Regex.
 
My first guess is that we operate on an disposed tree widget. However there is no widget disposed exception anywhere in the log file.

The second problem I ran into is that I cannot reproduce this (this only happens 1 out of 100 test runs) and so cannot debug the issue.

Can you at least ensure that the node exists in the Package Explorer view ? If not then this is an Eclipse issue.

File: stf/test/saros/stf/test/stf/contextmenu/ContextMenuShareProjectsTest.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    ALICE
+        .superBot()
+        .views()
+        .packageExplorerView()
+        .selectJavaProject(Constants.PROJECT1)
+        .openShareProjects();
+
+    IRemoteBotShell shell = ALICE.remoteBot().shell(SHELL_SHARE_PROJECT);
+    assertTrue(shell.activate());
```
Comment (srossbach) [inline stf/test/saros/stf/test/stf/contextmenu/ContextMenuShareProjectsTest.java:42]: https://github.com/saros-project/saros/blob/master/stf/src/saros/stf/server/rmi/remotebot/widget/impl/RemoteBotShell.java#L53  🤦

File: stf/test/saros/stf/test/stf/editor/EditorByAliceBobTest.java [hunk 2]
```diff
@@ -41,6 +45,11 @@ public void openAliceEditor() throws Exception {
     ALICE.remoteBot().editor(Constants.CLS1 + SUFFIX_JAVA).waitUntilIsActive();
   }
 
+  @After
+  public void tearDown() throws Exception {
+    leaveSessionPeersFirst(ALICE);
+  }
+
   @Test
   public void isEditorOpen() throws RemoteException {
```
Comment (srossbach) [inline stf/test/saros/stf/test/stf/editor/EditorByAliceBobTest.java:54]: Looking at the whole tests the original author supposed they were executed in the order they appeared.

The name of the whole test class says nothing, looking at the test code it test the follow mode. So basically you could merge all tests into one and put it to the test class that is currently testing some follow mode stuff.

File: stf/test/saros/stf/test/stf/editor/EditorByAliceTest.java [hunk 3]
```diff
@@ -119,21 +117,27 @@ public void autoComplateProposal() throws RemoteException {
...
+            .contains("public static void main"));
 
     ALICE
         .remoteBot()
         .editor(Constants.CLS1_SUFFIX)
         .autoCompleteProposal("main", "main - main method");
-    assertContains(
-        "public static void main",
-        ALICE.remoteBot().editor(Constants.CLS1_SUFFIX).getTextOnLine(3));
+    assertTrue(
```
Comment (srossbach) [inline stf/test/saros/stf/test/stf/editor/EditorByAliceTest.java:131]: Did the former fail of the assertion printed something like ("xyz was not found in abc") ? If yes then this is a drawback as it swallow the content of the content to check.

File: stf/test/saros/stf/test/stf/menubar/SarosPreferencesTest.java [hunk 4]
```diff
@@ -37,6 +24,10 @@ public void beforeEveryTest() throws Exception {
     ALICE.controlBot().getAccountManipulator().deleteAllAccounts();
   }
 
+  /*
```
Comment (srossbach) [inline stf/test/saros/stf/test/stf/menubar/SarosPreferencesTest.java:27]: Either delete them or set them to ignore. The class is named SarosPreferencesTest. These test cases do not test anything from the preference page.

**Comment by m273d15** [comment]
My contract at the FU Berlin ended. Therefore, I am not sure whether I find the time to integrate this change. For the time being, I close the PR but feel free to re-open it.

**Pull request closed without merging**